src/app/header/header.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, OnInit, Input, NgModule } from '@angular/core';
+import { UpdateDataService } from '../update-data.service';
+import { ArticleFormat } from '../format';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+
+export class HeaderComponent implements OnInit {
+  headerLabel = "Welcome to NewsFeeds";
+  filterLabel = "";
+  sourcesSet = new Set();
+  show:boolean = true;
```

Comment:
`show` isn't a good variable name. Rename it to a meaningful one which represents it's purpose

Reply:
Variable name changed to navbarHidingFlag